### PR TITLE
drogon 1.9.0 (new formula)

### PR DESCRIPTION
**Pull request by ken-matsui**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/d/drogon.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  test do
+    system bin/"dg_ctl", "create", "project", "hello"
+    cd "hello" do
+      system "cmake", "-B", "build"
+      system "cmake", "--build", "build"
+
+      begin
+        pid = fork { exec "build/hello" }
+        sleep 1
+        result = shell_output("curl -s 127.0.0.1:5555")
```
Comment (SMillerDev) [inline Formula/d/drogon.rb:40]: Can we pick a specific port here?

Comment (ken-matsui) [reply Formula/d/drogon.rb:40]: The port is currently hardcoded to `5555`. Would it be better for us to be able to choose a port via something like a $PORT environmental variable? I can update the upstream if it is required to resolve this error.

Comment (ken-matsui) [reply Formula/d/drogon.rb:40]: It is only for new projects. Users can always change the port number by manually editing the generated `main.cc`.

https://github.com/drogonframework/drogon/blob/f215cb15a0f53abd0ca7ee8b95ed8c9c3b40d262/drogon_ctl/templates/demoMain.csp#L4

Comment (SMillerDev) [reply Formula/d/drogon.rb:40]: Let's do `port = free_port` and `inreplace "main.cc", "5555", port` then.

**Comment by chenrui333** [comment]
```
  ==> cmake -B build
  Failed to execute: cmake
  Error: drogon: failed
  An exception occurred within a child process:
```

File: Formula/d/drogon.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  desc "Modern C++ web application framework"
+  homepage "https://drogon.org"
+  # pull from git tag to get submodules
+  url "https://github.com/drogonframework/drogon.git",
+      tag:      "v1.9.0",
+      revision: "f215cb15a0f53abd0ca7ee8b95ed8c9c3b40d262"
+  license "MIT"
+  head "https://github.com/drogonframework/drogon.git", branch: "master"
+
+  depends_on "cmake" => :build
```
Comment (SMillerDev) [inline Formula/d/drogon.rb:11]:
```suggestion
  depends_on "cmake" => [:build, :test]
```

Comment (ken-matsui) [reply Formula/d/drogon.rb:11]: Thank you!

**Comment by ken-matsui** [comment]
@SMillerDev Thanks for your reviews!

File: Formula/d/drogon.rb [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  depends_on "cmake"
+  depends_on "jsoncpp"
+  depends_on "ossp-uuid"
+
+  uses_from_macos "zlib"
+
+  def install
+    cmake_args = std_cmake_args
+    if OS.linux?
+      cmake_args << "-DUUID_LIBRARIES=uuid"
```
Comment (SMillerDev) [inline Formula/d/drogon.rb:22]: Can these be detected if we depend on pkg-config?

Comment (ken-matsui) [reply Formula/d/drogon.rb:22]: Possibly, let me try it.

Comment (ken-matsui) [reply Formula/d/drogon.rb]: Seems not working. https://github.com/Homebrew/homebrew-core/actions/runs/6979802227/job/18993904524?pr=152693#step:4:157

**Review by chenrui333 — APPROVED** [review]
LGTM, thanks!

**Review by p-linnane — APPROVED** [review]

**Pull request merged**